File: 2024/March/3-13 2485 Find the Pivot Integer/PivotInteger.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        pLeft = 1
        pRight = n
        sumLeft = pLeft
        sumRight = pRight

        while pLeft <= pRight:
            if pLeft == pRight:
                if sumLeft == sumRight:
                    return pLeft
        
            if sumLeft < sumRight:
                pLeft += 1
                sumLeft += pLeft
            else:
                pRight -= 1
                sumRight += pRight
        
        return -1

File: 2024/March/3-13 2485 Find the Pivot Integer/PivotInteger.java
class Solution {
    public int pivotInteger(int n) {
        int pLeft = 1;
        int pRight = n;
        int sumLeft = pLeft;
        int sumRight = pRight;

        while(pLeft <= pRight) {
            if(pLeft == pRight) {
                if(sumLeft == sumRight) {
                    return pLeft;
                }
            }

            if(sumLeft < sumRight) {
                pLeft++;
                sumLeft += pLeft;
            }
            else {
                pRight--;
                sumRight += pRight;
            }
        }

        return -1;
    }
}